code/create_survey_geojson.py

```diff
@@ -0,0 +1,152 @@
+
+    Parameters
+    ----------
+    survey_states_and_clusters : Iterable[str]
+        State and state cluster names to include in the GeoJSON file. State
+        cluster names should be in the format "State1, State2, ... (Cluster X)".
+    fpath_states_json : Path | str, optional
+        Path to GeoJSON file with standard US states.
+        By default, will use the one that was
+        downloaded from https://github.com/PublicaMundi/MappingAPI/blob/master/data/geojson/us-states.json.
+    fpath_out : Path | str, optional
+        Path to output GeoJSON file.
+
+    Returns
+    -------
+    dict
+        GeoJSON of the survey states and state clusters.
+
+    Raises
+    ------
+    RuntimeError
+        Raised if state clusters cannot be parsed.
+    """
+    # use default paths if none provided
+    if fpath_states_json is None:
+        fpath_states_json = FPATH_STATES_JSON_DEFAULT
+    if fpath_out is None:
+        fpath_out = FPATH_OUT_DEFAULT
+
+    fpath_states_json = Path(fpath_states_json)
+    fpath_out = Path(fpath_out)
+
+    # load GeoJSON file
+    states_json = json.loads(fpath_states_json.read_text())
+
+    # create mapping for easy access to reference state information
+    state_info_map = {
+        info["properties"]["name"]: info for info in states_json["features"]
+    }
+
+    # split into states and clusters
+    survey_states = [
+        state for state in survey_states_and_clusters if "Cluster" in state
+    ]
+    survey_clusters = [
+        state
+        for state in survey_states_and_clusters
+        if state not in survey_states
+    ]
+
+    survey_state_info = []
+
+    # add single states
+    for state in survey_clusters:
+        print(f"Adding state {state}")
+        state_info = state_info_map[state]
+        state_info["id"] = state
+        del state_info["properties"]["density"]
+        survey_state_info.append(state_info)
+
+    # add clusters
+    for cluster in survey_states:
+
+        # get individual state names
+        try:
+            clustered_states = (
+                re.search("(.*) \\(Cluster", cluster).groups()[0].split(", ")
+            )
+        except Exception as exception:
+            raise RuntimeError(
+                f"Could not parse states in cluster {cluster}: {exception}"
+            )
+
+        print(f"Adding cluster {cluster}")
+        cluster_info = None
+        for clustered_state in clustered_states:
+            print(f"\tAdding state {clustered_state}")
+
+            # first state in the cluster
+            if cluster_info is None:
+                cluster_info = state_info_map[clustered_state]
+                cluster_info["properties"]["name"] = cluster
+                cluster_info["id"] = cluster
+
+                # convert into MultiPolygon if needed
+                if cluster_info["geometry"]["type"] == "Polygon":
+                    cluster_info["geometry"]["type"] = "MultiPolygon"
```

Comment:
purely for my own understanding: if we wanted to remove the state boundaries (i.e., distinction b/w state polygons) when doing this merging, we would combine them using a `Polygon` instead, is that right?

to be clear, I still think having the within-cluster state lines is desired in our case. just interested :)

Reply:
So I think we would have to do some sort of reduction of the polygon (i.e. apply an algorithm that says: "here are a bunch of polygons, give me the smallest number of polygons satistying some criteria"). Some of the states/clusters would still have to be `MultiPolygon`s because they have islands.